src/components/Form/InputWrapper.js

```diff
@@ -17,8 +19,9 @@ const defaultProps = {
 function InputWrapper(props) {
     const {InputComponent, inputID, forwardedRef, ...rest} = props;
     const {registerInput} = useContext(FormContext);
+    const shouldSetTouchedOnBlurOnly = InputComponent === TextInput;
```

Comment:
Can you please add a code comment to explain why `set touched` is different for `InputComponent` and `TextInput`?